OpenSim/Common/DataTable.h

```diff
@@ -0,0 +1,281 @@
+    DataTable_()                             = default;
+    DataTable_(const DataTable_&)            = default;
+    DataTable_(DataTable_&&)                 = default;
+    DataTable_& operator=(const DataTable_&) = default;
+    DataTable_& operator=(DataTable_&&)      = default;
+    ~DataTable_()                            = default;
+
+    std::unique_ptr<AbstractDataTable> clone() const override {
+        return std::unique_ptr<AbstractDataTable>{new DataTable_{*this}};
+    }
+
+    /** Append row to the DataTable_.                                         */
+    void appendRow(const ETX& indRow, const RowVector& depRow) {
+        validateRow(_indData.size(), indRow, depRow);
+
+        _indData.push_back(indRow);
+
+        if(_depData.nrow() == 0 || _depData.ncol() == 0) {
+            try {
+                auto& labels = 
+                    _dependentsMetaData.getValueArrayForKey("labels");
+                if(depRow.size() != labels.size())
+                    throw Exception{"Number of columns in the input row does "
+                            "not match the number of labels."};
+            } catch(std::out_of_range&) {
+                // No operation.
+            }
+            _depData.resize(1, depRow.size());
+        } else
+            _depData.resizeKeep(_depData.nrow() + 1, _depData.ncol());
+            
+        _depData.updRow(_depData.nrow() - 1) = depRow;
+    }
+
+    /** Get row at index.                                                     */
+    RowVectorView getRowAtIndex(size_t index) const {
+        return _depData.row(index);
+    }
+
+    /** Get row corresponding to the given entry in the independent column.   */
+    RowVectorView getRow(const ETX& ind) const {
+        auto iter = std::find(_indData.cbegin(), _indData.cend(), ind);
+
+        if(iter == _indData.cend())
+            throw Exception{"Independent column has no entry with value " +
+                    std::to_string(ind)};
+
+        return _depData.row(std::distance(_indData.cbegin(), iter));
+    }
+
+    /** Update row at index.                                                  */
+    RowVectorView updRowAtIndex(size_t index) {
+        return _depData.updRow(index);
+    }
+
+    /** Update row corresponding to the given entry in the independent column.*/
+    RowVectorView updRow(const ETX& ind) {
+        auto iter = std::find(_indData.cbegin(), _indData.cend(), ind);
+
+        if(iter == _indData.cend())
+            throw Exception{"Independent column has no entry with value " +
+                    std::to_string(ind)};
+
+        return _depData.updRow(std::distance(_indData.cbegin(), iter));
+    }
+
+    /** Get independent column.                                               */
+    const std::vector<ETX>& getIndependentColumn() const {
+        return _indData;
+    }
+
+    /** Get dependent column.                                                 */
+    ColumnVector getDependentColumnAtIndex(size_t index) const {
```

Comment:
Did you want this to be a copy of the column, @klshrinidhi, or should it be using a `VectorView`?